lib/spack/spack/relocate.py

```diff
@@ -390,16 +395,18 @@ def _set_elf_rpaths(target, rpaths):
     """Replace the original RPATH of the target with the paths passed
     as arguments.
 
-    This function uses ``patchelf`` to set RPATHs.
-
     Args:
         target: target executable. Must be an ELF object.
         rpaths: paths to be set in the RPATH
 
     Returns:
         A string concatenating the stdout and stderr of the call
-        to ``patchelf``
+        to ``patchelf`` if it was invoked
     """
+    # Try a fast in-place method before trying patchelf
+    if elf.replace_rpaths(target, rpaths, force=elf.RPathType.RPATH):
+        return ''
```

Comment:
IMO it's a bit much to combine (a) a check for missing libraries and also (b) to add an optimization for setting RPATHs. Minimally I think these should be managed as two commits (in which case the actual commit content would matter since I would be rebasing instead of squashing) but my preference is separate PRs. Let me know if I'm missing something and this is essential to the goal of detecting missing libraries.

Reply:
It's from another PR: #27610